pkg/coprocessor/observer.go

```diff
@@ -0,0 +1,94 @@
+package coprocessor
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"github.com/grafana/loki/pkg/coprocessor/proto"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/grafana/loki/pkg/loghttp"
+	lokiutil "github.com/grafana/loki/pkg/util"
+	"github.com/grafana/loki/pkg/util/build"
+)
+
+var timeout = 2 * time.Minute
```

Comment:
I think this should be passed as a config along the URL.

Reply:
thanks.done